openassessment/workflow_batch_update_api.py

```diff
@@ -0,0 +1,269 @@
+"""
+Provides functionality to batch update ORA workflows for different scopes
+"""
+
+import logging
+import time
+import datetime
+from django.utils import timezone
+
+from opaque_keys.edx.keys import UsageKey
+from openassessment.runtime_imports.functions import modulestore
+from openassessment.assessment.models import PeerWorkflow
+from openassessment.workflow import api
+
+logger = logging.getLogger(__name__)
+
+
+def update_workflows_for_ora_block(item_id):
```

Comment:
My understanding was that these would be a chain of functions that each put more tasks on the celery queue. My understanding of the high-level flow was:

```
- Query for which ORAs had learners that were waiting for a peer review for more than 7 days
- for each course, create a celery task, passing in the course id and list of potential
 ora blocks as params
- that celery task would load the course block and check if the course-wide flexible 
override was enabled, then for each ora item it was given, put a new celery task on 
the queue with the override value and the item id as parameters
- that celery task would load the specified ora block. if the course override and the
 block-level setting for flexible grading were both disabled, noop, otherwise, look 
 up all waiting 7 day old workflows for this block id and put individual celery tasks 
 on the queue with a workflow identifier and the block's assessment requirements
- that celery task would then run the individual workflow update.
```

Why did this architecture change? Here we have the multiple levels, but not really in a chaining async task way, but in the sense of "choose which context you want to do this in" and then we do all work within the one worker. updating multiple courses within a single celery task is too much for a single worker to do. 

I know we discussed potentially "squashing" one or two of those "layers" above into one layer, but I think the current approach goes a bit too far. The work needs to be split up more.